zend_parse_arg_impl.go

```diff
@@ -73,3 +66,52 @@ func zendParseArgAsString(value any) (string, error) {
 	}
 	return str, nil
 }
+
+// ConvertToString attempts to convert the given value to string, emulating PHP 5.6'S _convert_to_string behavior.
+// It supports basic Go types and handles special cases for *os.File, net.Conn, and *sql.DB and types implementing toString()
+// Unlike PHP, which has built-in support for managing resource IDs for types like files and database connections,
+// Go does not inherently manage resource IDs. Due to this language difference, this function uses the values' pointer
+// address as the pseudo resource ID for identifiable resource types.
+//
+// Reference:
+//   - _convert_to_string implementation:
+//     https://github.com/php/php-src/blob/php-5.6.40/Zend/zend_operators.c#L593-L661
+//   - convert_object_to_type implementation:
+//     https://github.com/php/php-src/blob/php-5.6.40/Zend/zend_operators.c#L333-L357
+func ConvertToString(value any) (string, error) {
+	if value == nil {
+		return "", nil
+	}
+
+	// use reflection to handle basic and composite types dynamically
+	v := reflect.ValueOf(value)
+	switch v.Kind() {
```

Comment:
I'm not sure how more efficient it is, but because generally the cost of reflection is high, I think it would be better to use type assertion for primitive types, interface, struct, or map things

```go
// go build -o assertion main.go
// go tool objdump assertion
// And then see "TEXT main.main" part
func main() {
    var val interface{} = complex128(100)

    switch val.(type) {
    case string:
        fmt.Println("string type")
    case int:
        fmt.Println("int type")
    case int16:
        fmt.Println("int16 type")
    case int32:
        fmt.Println("int32 type")
    case int64:
        fmt.Println("int64 type")
    case uint:
        fmt.Println("uint type")
    case uint8:
        fmt.Println("uint8 type")
    case uint16:
        fmt.Println("uint16 type")
    case uint32:
        fmt.Println("uint32 type")
    case uint64:
        fmt.Println("uint64 type")
    case float32:
        fmt.Println("float32 type")
    case float64:
        fmt.Println("float64 type")
    case complex64:
        fmt.Println("complex64 type")
    case complex128:
        fmt.Println("complex128 type")
    case bool:
        fmt.Println("bool type")
    default:
        fmt.Println("unknown type")
    }
}
```

```go
// go build -o reflection main.go
// go tool objdump reflection
// And then see "TEXT main.main" part
func main() {
var val interface{} = complex128(100)

switch reflect.ValueOf(val).Kind() {
case reflect.String:
fmt.Println("string type")
case reflect.Int:
fmt.Println("int type")
case reflect.Int16:
fmt.Println("int16 type")
case reflect.Int32:
fmt.Println("int32 type")
case reflect.Int64:
fmt.Println("int64 type")
case reflect.Uint:
fmt.Println("uint type")
case reflect.Uint8:
fmt.Println("uint8 type")
case reflect.Uint16:
fmt.Println("uint16 type")
case reflect.Uint32:
fmt.Println("uint32 type")
case reflect.Uint64:
fmt.Println("uint64 type")
case reflect.Float32:
fmt.Println("float32 type")
case reflect.Float64:
fmt.Println("float64 type")
case reflect.Complex64:
fmt.Println("complex64 type")
case reflect.Complex128:
fmt.Println("complex128 type")
case reflect.Bool:
fmt.Println("bool type")
default:
fmt.Println("unknown type")
}
}
```

When I compare objdump result of two code snippets, the former has fewer lines than latter.
And the result of objdump of reflection shows that `reflect` package do more memory related things and more complex things than type assertion.

So, for [primitive types](https://go.dev/ref/spec#Types), I think it would be better to use type assertion than reflection.